File: ChompMan/DataAccess/IScoreRepository.vb
Imports System.Collections.Generic

Namespace DataAccess

    ''' <summary>
    ''' Contract for high-score storage.
    ''' Implementations may back this with Access, SQL Server, EF Core, etc.
    ''' </summary>
    Public Interface IScoreRepository

        ''' <summary>Returns the top <paramref name="count"/> scores, ordered descending.</summary>
        Function GetTopScores(Optional count As Integer = 10) As List(Of ScoreEntry)

        ''' <summary>
        ''' Persists a new high-score row.
        ''' Creates the player record if the name has not been seen before.
        ''' </summary>
        Sub SaveScore(playerName As String, score As Integer, levelReached As Integer)

    End Interface

End Namespace

File: ChompMan.Core/DataAccess/IScoreRepository.cs
namespace ChompMan.DataAccess;

/// <summary>
/// Contract for high-score storage.
/// Implementations may back this with Access, SQL Server, EF Core, etc.
/// </summary>
public interface IScoreRepository
{
    /// <summary>Returns the top <paramref name="count"/> scores, ordered descending.</summary>
    List<ScoreEntry> GetTopScores(int count = 10);

    /// <summary>
    /// Persists a new high-score row.
    /// Creates the player record if the name has not been seen before.
    /// </summary>
    void SaveScore(string playerName, int score, int levelReached);
}
